meos/src/general/meos.c

```diff
@@ -474,8 +477,60 @@ meos_finalize(void)
   meos_finalize_timezone();
   /* Finalize PROJ */
   proj_context_destroy(_PJ_CONTEXT);
+  /* Finalize GSL */
+  gsl_rng_free(_GENERATION_RNG);
+  gsl_rng_free(_AGGREGATION_RNG);
```

Comment:
This should maybe go into a `gsl_finalize()` function that also sets `_GSL_INITIALIZED` to `false`.